probe/endpoint/connection_tracker.go

```diff
@@ -87,7 +91,27 @@ func (t *connectionTracker) ReportConnections(rpt *report.Report) {
 			t.performEbpfTrack(rpt, hostNodeID)
 			return
 		}
-		log.Warnf("ebpf tracker died, gently falling back to proc scanning")
+
+		// We only restart the EbpfTracker if the failures are not too frequent to
+		// avoid repeatitive restarts.
+		if t.ebpfFailureCount > 0 && t.ebpfFailureTime.Before(time.Now().Add(-5*time.Minute)) {
+			t.ebpfFailureCount = 0
+		}
+		t.ebpfFailureTime = time.Now()
+
+		if t.ebpfFailureCount == 0 {
```

Comment:
This logic is hard to read. I don't think there's a need to split this in multiple `if`s. I think it's preferable to repeat the `t.useProcfs()`. Also, it's more idiomatic in Go to check for errors in `if`s and continue with the least erroneous path.

```
t.ebpfFailureTime = time.Now()
// Multiple failures in the last 5 minutes, fall back to proc parsing
if t.ebpfFailureCount > 0 && t.ebpfFailureTime.After(time.Now().Add(-5*time.Minute))  {
   log.Warnf("ebpf tracker died again, gently falling back to proc scanning")
   t.useProcfs()
  return
}

// Tolerable failure rate, let's just reset the failure count and restart the traker
t.ebpfFailureCount = 0
log.Warnf("ebpf tracker died, restarting it")
if err := t.ebpfTracker.restart(); err != nil {
  log.Warnf("could not restart ebpf tracker, falling back to proc scanning: %v", err)
  t.useProcfs()
  return
}

go t.getInitialState()
t.performEbpfTrack(rpt, hostNodeID)
return
```